app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuth.scala

```diff
@@ -120,6 +122,21 @@ trait HelpToSaveAuth extends AuthorisedFunctions with AuthRedirects with Logging
       }
     }
 
+  private def isInMaintenance(currentTime: Long, maintenanceTimes: Seq[appConfig.MaintainceTimes]): Boolean = {
+    val currentDateTime =
+      if (isSummerTime(LocalDateTime.now())) LocalDateTime.now().minusHours(1) else LocalDateTime.now()
+    val checkTimes = maintenanceTimes.map(
+      mt ⇒
+        if (currentDateTime.isAfter(mt.startTime) && currentDateTime.isBefore(mt.endTime)) {
+          endTime = mt.endTime.toString()
```

Comment:
No, you should not use a var to do that!

Why not have isMaintenance instead be a method that passes out an option of an end time, then you can pass that as the parameter of the MaintenanceException?